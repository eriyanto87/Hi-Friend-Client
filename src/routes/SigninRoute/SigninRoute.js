import React from "react";
import Signin from "../../components/Signin/Signin";
import Home from "../../components/Home/Home";

export default function SigninRoute() {
  return (
    <>
      <Home />
      <Signin />
    </>
  );
}
