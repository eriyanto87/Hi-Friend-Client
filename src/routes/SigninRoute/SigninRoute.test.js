import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import SigninRoute from "./SigninRoute";

describe("Sign in component", () => {
  it("Sign in Snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SigninRoute />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <SigninRoute />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
});
