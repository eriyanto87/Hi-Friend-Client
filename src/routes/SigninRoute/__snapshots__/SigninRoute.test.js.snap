// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in component Sign in Snapshot 1`] = `
Array [
  <div
    className="home"
  >
    <div>
      <a
        href="/"
      >
        Go To Home
      </a>
    </div>
  </div>,
  <div
    className="signin-container"
  >
    <section
      className="signin-header"
    >
      <h1>
        Sign In Form
      </h1>
      <p>
         
        Username: Demo 
        <br />
        Password: Password123!
      </p>
    </section>
    <form
      onSubmit={[Function]}
    >
      <section
        className="signin-form"
      >
        <p>
          <input
            name="username"
            placeholder="Username"
            required={true}
            type="text"
          />
        </p>
        <p>
          <input
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
        </p>
      </section>
      <button>
        Sign In
      </button>
    </form>
  </div>,
  <p
    className="signin-section"
  >
    Don't have an account? 
    <a
      href="/registration"
    >
      Register Here
    </a>
  </p>,
  <div
    className="footer"
  >
    Â© 2020
  </div>,
]
`;
